orderer/common/cluster/deliver.go

```diff
@@ -100,7 +106,13 @@ func (p *BlockPuller) PullBlock(seq uint64) *common.Block {
 			p.Logger.Errorf("Failed pulling block [%d]: retry count exhausted(%d)", seq, p.MaxPullBlockRetries)
 			return nil
 		}
-		time.Sleep(p.RetryTimeout)
+
+		select {
```

Comment:
This is a recurring pattern (both here and below) so can we not simply extract it to a method, and then add in the for loop a condition of isClosed() that simply probes if the channel is closed? Much more appealing to the eye

Reply:
Done